ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh

```diff
@@ -24,6 +45,7 @@ if [[ "$SINGLE_ZONE" == "true" ]]; then
         fi
     done
 else
+    node_zone_list=$(oc get nodes -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}')
```

Comment:
Per the script logic, when compute or control plane are set custom zone, and run the above zone checking, still run this default zone checking ?

Reply:
I should focus on it. IMO unless we set both `COMPUTE_ZONE` and `CONTROL_PLANE_ZONE`, we should do default zone checking, what about change 
`else` 
to
`elif [[ "$COMPUTE_ZONE" == "" ]] || [[ "$CONTROL_PLANE_ZONE" == "" ]]; then`

that means we should do default check if only set one of `COMPUTE_ZONE` and `CONTROL_PLANE_ZONE`

Reply:
I think we can refactor the default checking way, the logic would be like this:
```
if [[ "$SINGLE_ZONE" == "true" ]]; then
    ....
else
    default_node_pes=$(echo "$prism_element1" "$prism_element2" "$prism_element3" | tr " " "\n" | sort -u | xargs)
    control_plane_pes="$default_node_pes"
    compute_pes="$default_node_pes"
    compute_zone_list=$(oc get nodes -l node-role.kubernetes.io/worker= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)
    control_plane_zone_list=$(oc get nodes -l node-role.kubernetes.io/master= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)
    if [[ "$CONTROL_PLANE_ZONE" != "" ]]; then
        control_plane_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
    fi
    if [[ "$COMPUTE_ZONE" != "" ]]; then
        compute_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
    fi
    if [[ $control_plane_zone_list == "$control_plane_pes" ]]; then
        echo "Pass: passed to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
    else
        echo "Fail: fail to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
        exit 1
    fi
    if [[ $compute_zone_list == "$compute_pes" ]]; then
        echo "Pass: passed to check compute zone: $compute_zone_list, expected: $compute_pes"
    else
        echo "Fail: fail to check compute zone: $compute_zone_list, expected: $compute_pes"
        exit 1
    fi
```

Reply:
Only one thing, should we consider compute nums less than 3? What about 
`if [[ $compute_zone_list == "$compute_pes" ]]; then`
change to
`if [[ $compute_pes == "$compute_zone_list" ]]; then`

Reply:
Now we have 3 failure domains, "$prism_element1" "$prism_element2" "$prism_element3", when computes number less than 3, the compute nodes would random select the failure domains, might be  "$prism_element1"  and  "$prism_element3", right?

Reply:
And another point is we change to use a reg match, if the compute number is >=3, if some bug happened, all the computes are landed to "$prism_element1", the reg match would not catch the issue, right?

Reply:
yes, it will random select 2 from 3 failure domains

Reply:
I made some change like
```
    default_node_pes=$(echo "$prism_element1 $prism_element2 $prism_element3" | tr " " "\n" | sort -u | xargs)
    compute_zone_list=$(oc get nodes -l node-role.kubernetes.io/worker= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)
    control_plane_zone_list=$(oc get nodes -l node-role.kubernetes.io/master= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)

    if [[ "$COMPUTE_ZONE" != "" ]]; then
        compute_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
        if [[ $compute_zone_list == "$compute_pes" ]]; then
            echo "Pass: passed to check compute zone: $compute_zone_list, expected: $compute_pes"
        else
            echo "Fail: fail to check compute zone: $compute_zone_list, expected: $compute_pes"
            exit 1
        fi
    else
        #In case of compute nums less than 3, use =~
        if [[ $default_node_pes =~ $compute_zone_list ]]; then
            echo "Pass: passed to check compute defaul zone: $compute_zone_list, expected: $default_node_pes"
        else
            echo "Fail: fail to check compute defaul zone: $compute_zone_list, expected: $default_node_pes"
            exit 1
        fi
    fi

    if [[ "$CONTROL_PLANE_ZONE" != "" ]]; then
        control_plane_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
        if [[ $control_plane_zone_list == "$control_plane_pes" ]]; then
            echo "Pass: passed to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
        else
            echo "Fail: fail to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
            exit 1
        fi
    else
        if [[ $control_plane_zone_list == "$default_node_pes" ]]; then
            echo "Pass: passed to check control plane defaul zone: $control_plane_zone_list, expected: $default_node_pes"
        else
            echo "Fail: fail to check control plane defaul zone: $control_plane_zone_list, expected: $default_node_pes"
            exit 1
        fi
    fi
```

Reply:
If that, `if [[ $compute_pes =~ "$compute_zone_list" ]]; then` would be failed.

If we want to meet all cases, then need to judge the compute number number, when the compute number is >=3 or the compute number is <3, use different ways to run the check.